### PR TITLE
fix: dialogue cleanup

### PR DESCRIPTION
**Pull request by dvilelaf**
## Proposed changes

Create a simple dialogue cleanup method that drops every dialogue and label in storage

## Fixes

n/a

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: aea/protocols/dialogue/base.py [hunk 1]
```diff
@@ -1071,6 +1071,13 @@ def __init__(self, dialogues: "Dialogues") -> None:
         # used for both storing complete and incomplete dialogue labels
         self._terminal_state_dialogues_labels: Set[DialogueLabel] = set()
 
+    def cleanup(self) -> None:
```
Comment (DavidMinarsch) [inline aea/protocols/dialogue/base.py:1074]: I just noticed that BaseDialoguesStorage has a dependency on Dialogues which means there is circularity. Can we break this? Quick check seems to suggest we don't need self._dialogues here (apart from a single flag, which can be provided as an argument to BaseDialoguesStorage instead).

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]
Where are the tests?

File: aea/protocols/dialogue/base.py (hunk 1 above)
Comment (dvilelaf) [reply aea/protocols/dialogue/base.py:1074]: `PersistDialoguesStorage` (derived from `BasicDialoguesStorage`) uses `self._dialogues`  in several places, but I'll try to break the dependency.

Comment (DavidMinarsch) [reply aea/protocols/dialogue/base.py:1079]: I think it would be better to try this on a separate PR and roll back the changes here!

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**